Makefile

```diff
@@ -1,131 +1,44 @@
+.PHONY: help build build_repro build_debug build_wrapper clean
 
 help:
 	@awk 'BEGIN {FS = ":.*?## "} /^[a-zA-Z0-9_-]+:.*?## / {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}' $(MAKEFILE_LIST)
 
-build: build_compiled ## build default container image (production)
-
-build_environment: ## build environment container image
-	docker build -f $(DOCKERFILE) -t $(ENVIRONMENT_IMAGE) \
-	--target $(ENVIRONMENT_TARGET) .
-
-build_compiled: ## build production container image
-	docker build \
-		--build-arg compile_option=$(COMPILE_OPTION) \
-		$(BUILD_ARGS) \
-		-f $(DOCKERFILE) \
-		-t $(COMPILED_IMAGE) \
-		--target $(COMPILE_TARGET) .
-
-build_debug: ## build container image for debugging
-	COMPILED_TAG_NAME=$(COMPILED_TAG_NAME)-debug COMPILE_OPTION="REPRO=\\\nDEBUG=Y" $(MAKE) build
-
-build_serialize: ## build container image for serialization
-	BUILD_ARGS="$(BUILD_ARGS) --build-arg serialize=true" \
-	COMPILED_IMAGE=$(SERIALIZE_IMAGE) \
-	$(MAKE) build_compiled
-
-build_deps: ## build container images of dependnecies (FMS, ESMF, SerialBox)
-	docker build -f $(DOCKERFILE) -t $(MPI_IMAGE) $(BUILD_ARGS) --target fv3gfs-mpi .
-	docker build -f $(DOCKERFILE) -t $(FMS_IMAGE) $(BUILD_ARGS) --target fv3gfs-fms .
-	docker build -f $(DOCKERFILE) -t $(ESMF_IMAGE) $(BUILD_ARGS) --target fv3gfs-esmf .
-	docker build -f $(DOCKERFILE) -t $(SERIALBOX_IMAGE) $(BUILD_ARGS) --target fv3gfs-environment-serialbox .
-
-push_image_%:
-	docker tag $(GCR_URL)/$*:$(DEP_TAG_NAME) $(GCR_URL)/$*:$(DEP_TAG_NAME)-$(COMMIT_SHA)
-	docker push $(GCR_URL)/$*:$(DEP_TAG_NAME)
-	docker push $(GCR_URL)/$*:$(DEP_TAG_NAME)-$(COMMIT_SHA)
-
-## push container images of dependencies to GCP 
-push_deps: push_image_mpi-build push_image_fms-build push_image_esmf-build push_image_serialbox-build
-
-pull_deps: ## pull container images of dependencies from GCP (for faster builds)
-	docker pull $(MPI_IMAGE)
-	docker pull $(FMS_IMAGE)
-	docker pull $(ESMF_IMAGE)
-	docker pull $(SERIALBOX_IMAGE)
-
-enter: ## run and enter production container for development
-	docker run --rm \
-		-v $(shell pwd)/FV3:/FV3 $(OTHER_MOUNTS) \
-		-w /FV3 -it $(COMPILED_IMAGE) bash
-
-enter_serialize: ## run and enter serialization container for development
-	docker run --rm \
-		-v $(shell pwd)/FV3:/FV3/original $(OTHER_MOUNTS) \
-		-w /FV3 -it $(SERIALIZE_IMAGE) bash
-
-test: ## run tests (set COMPILED_TAG_NAME to override default)
-	pytest tests/pytest --capture=no --verbose --refdir $(shell pwd)/tests/pytest/reference/circleci --image_version $(COMPILED_TAG_NAME)
-
-build_native: ## build FV3 locally (assuming all tools and dependencies are available in the environment)
+build: ## build repro and debug mode versions FV3 locally (assuming all tools and dependencies are available in the environment)
+	$(MAKE) build_repro
+	$(MAKE) clean_fortran_build_artifacts
+	$(MAKE) build_debug
+
+build_repro: ## build repro mode version of FV3 locally (assuming all tools and dependencies are available in the environment)
```

Comment:
I've done a quick scan of the code changes and I think I'm following along, but I'd like to give a try at checking out the branch and trying the build/test process tomorrow, since I might provide more useful feedback that way.